### PR TITLE
Fix for latest Observium CE and use with hostname with dashes

### PR DESCRIPTION
**Pull request by nicodier**

### CODE REVIEW

File: graph.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def sizeof_fmt(num, suffix='bps'):
 
 def discover_links(graph, this_device):
     global debug_id
-    cursor.execute(query_template, this_device)
+    cursor.execute(query_template % this_device)
```
Comment (sysblade) [inline graph.py:143]: Why this change ?
Using parameter instead of string interpolation protects from potential SQL injections.

**Review by sysblade — COMMENTED** [review]
Thanks for the PR, I just added a comment about a specific change that I don't understand. the reason.

File: graph.py (hunk 1 above)
Comment (nicodier) [reply graph.py:143]: Because with ',' I get:

Generating graph : Traceback (most recent call last):
  File "/opt/observium/graph.py", line 191, in <module>
    graph = discover_links(graph, this_device)
  File "/opt/observium/graph.py", line 145, in discover_links
    cursor.execute(query_template, this_device)
  File "/usr/lib64/python2.7/site-packages/MySQLdb/cursors.py", line 187, in execute
    query = query % tuple([db.literal(item) for item in args])
TypeError: not all arguments converted during string formatting

I know it is not best practice, but at least it makes this thing work for now. If anyone wants to make a better version, feel free :)

Comment (sysblade) [reply graph.py:143]: Have you tried with `cursor.execute(query_template, str(this_device))` and removing the `'` you added in the SQL query ?
(I don't have any working Observium setup to test)

Comment (nicodier) [reply graph.py:143]: Can't get that to work. Get the same error.

Comment (sysblade) [reply graph.py:143]: Ok let's do your way until we find a better fix
